client/extensions/woocommerce/woocommerce-services/views/packages/add-package.js

```diff
@@ -0,0 +1,245 @@
+					value={ name || '' }
+					onChange={ updateTextField }
+					isError={ modalErrors.name }
+				/>
+				{ fieldInfo( 'name', translate( 'This field must be unique' ) ) }
+			</FormFieldset>
+			<FormFieldset>
+				<FormLabel>{ translate( 'Inner Dimensions (L x W x H) %(dimensionUnit)s', { args: { dimensionUnit } } ) }</FormLabel>
+				<FormTextInput
+					name="inner_dimensions"
+					placeholder={ exampleDimensions }
+					value={ inner_dimensions || '' }
+					onChange={ updateTextField }
+					disabled={ ! is_user_defined }
+					isError={ modalErrors.inner_dimensions }
+				/>
+				{ fieldInfo( 'inner_dimensions' ) }
+				{ ! isOuterDimensionsVisible ? <OuterDimensionsToggle { ...{ siteId, toggleOuterDimensions, translate } } /> : null }
+			</FormFieldset>
+			{ isOuterDimensionsVisible
+				? ( <FormFieldset>
+						<FormLabel>
+							{ translate( 'Outer Dimensions (L x W x H) %(dimensionUnit)s', { args: { dimensionUnit } } ) }
+						</FormLabel>
+						<FormTextInput
+							name="outer_dimensions"
+							placeholder={ exampleDimensions }
+							value={ outer_dimensions || '' }
+							onChange={ updateTextField }
+							disabled={ ! is_user_defined }
+							isError={ modalErrors.outer_dimensions }
+						/>
+						{ fieldInfo( 'outer_dimensions' ) }
+					</FormFieldset> )
+				: null
+			}
+			<FormFieldset className="packages__add-package-weight-group">
+				<div className="packages__add-package-weight">
+					<FormLabel htmlFor="box_weight">{ translate( 'Package weight' ) }</FormLabel>
+					<FormTextInput
+						id="box_weight"
+						name="box_weight"
+						placeholder={ translate( 'Weight of box' ) }
+						value={ box_weight || '' }
+						onChange={ updateTextField }
+						disabled={ ! is_user_defined }
+						isError={ modalErrors.box_weight }
+					/>
+					{ fieldInfo( 'box_weight' ) }
+				</div>
+				<div className="packages__add-package-weight">
+					<FormLabel htmlFor="max_weight">{ translate( 'Max weight' ) }</FormLabel>
+					<FormTextInput
+						id="max_weight"
+						name="max_weight"
+						placeholder={ translate( 'Max weight' ) }
+						value={ max_weight || '' }
+						onChange={ updateTextField }
+						disabled={ ! is_user_defined }
+						isError={ modalErrors.max_weight }
+					/>
+					<span className="packages__add-package-weight-unit">{ weightUnit }</span>
+					{ fieldInfo( 'max_weight' ) }
+				</div>
+				<FormSettingExplanation>
+					{ translate( 'Defines both the weight of the empty box and the max weight it can hold' ) }
+				</FormSettingExplanation>
+			</FormFieldset>
+		</Dialog>
+	);
+};
+
+AddPackageDialog.propTypes = {
+	siteId: PropTypes.number,
+	dismissModal: PropTypes.func.isRequired,
+	form: PropTypes.object.isRequired,
+	updatePackagesField: PropTypes.func.isRequired,
+	showOuterDimensions: PropTypes.bool,
+	toggleOuterDimensions: PropTypes.func.isRequired,
+	savePackage: PropTypes.func.isRequired,
+	packageData: PropTypes.object,
```

Comment:
Should probably enforce the parts of the shape that we rely on (e.g. index, name, etc)